overlord/snapstate/snapmgr.go

```diff
@@ -64,12 +64,17 @@ func (ss *SnapSetup) DevMode() bool {
 	return ss.Flags&int(snappy.DeveloperMode) != 0
 }
 
+func (ss *SnapSetup) TryMode() bool {
+	return ss.Flags&int(snappy.TryMode) != 0
```

Comment:
The leakage of snappy.* flags here is very unfortunate and super confusing. I didn't realize we were doing that.

It's okay for this branch, though. Let's fix both at once.